toolchain/sem_ir/formatter.cpp

```diff
@@ -1161,6 +1161,14 @@ class Formatter {
         .print(out_, sem_ir_.types().IsSignedInt(inst.type_id));
   }
 
+  auto FormatInstructionRHS(FloatLiteral inst) -> void {
+    // Arbitrarily uses 5 digits of precision when formatting
+    // float literals.
+    llvm::SmallVector<char, 16> buffer;
+    sem_ir_.floats().Get(inst.float_id).toString(buffer, 5);
```

Comment:
Does LLVM provide a way to produce a shortest representation that correctly round-trips? In principle we'd like our formatted SemIR to be parseable back to an equivalent file, so we should use a lossless printing mechanism here.

Reply:
It does. If we keep the precision argument at 0 (the default), LLVM will [use the full precision of the `APFLoat`](https://github.com/llvm/llvm-project/blob/08a787c91be51e6a561dc50ec34ce28abc04ac54/llvm/include/llvm/ADT/APFloat.h#L461) in it [will correctly roundtrip](https://github.com/llvm/llvm-project/blob/aac4d03423dd6b7bdef0f2eb03c570f3e2ca6630/llvm/lib/Support/APFloat.cpp#L4089).

Just updated the formatter to use that, meaning floats will now round-trip. The only caveat is "visual", since we now show the full precision (e.g. `0.10000000000000001` for `1/10`, such as [here](https://github.com/ppaglilla/carbon-lang/blob/9c4d94e0a12dd2d3fa80ae7e7265040e1c123476/toolchain/check/testdata/basics/builtin_types.carbon#L16)).